test/Table_Tests/src/Common_Table_Operations/Column_Operations_Spec.enso

```diff
@@ -1734,6 +1747,28 @@ add_specs suite_builder setup =
             t = table_builder [["x", ["1", "2", "3"]]]
             t.at "x" . const (t.at "x") . should_fail_with Illegal_Argument
 
+    if setup.is_database then
+        suite_builder.group prefix+"(Column_Operations_Spec) Column.let" group_builder->
+            group_builder.specify "Should allow the use of let-style CTEs" <|
+                t = table_builder [["X", [1, 2, 3, 4]], ["Y", [4, 3, 100, 200]], ["Z", [12.1, 23.4, -4.5, 44.33]]]
+                x = t.at "X"
+                y = t.at "Y"
+                z = t.at "Z"
+                a = ((x - 2.3) * (y + 1)) + z
+                b = (x - 2.3).let name="xx" xx-> (xx * (y + 1)) + z
+                c = (x - 2.3).let name="xx" xx-> (y + 1).let name="yy" yy-> (xx * yy) + z
+                d = ((x - 2.3) * (y + 1)).let name="xxyy" xxyy-> xxyy + z
+                a.to_vector . should_equal [5.6000000000000005, 22.2, 66.20000000000002, 386.03000000000003]
+                a.to_vector . should_equal b.to_vector
+                a.to_vector . should_equal c.to_vector
+                a.to_vector . should_equal d.to_vector
```

Comment:
This will run 7 queries overall. I'd rather decrease the overhead. At least save `a.to_vector` to a variable instead of calling each time.

Even better - use `set` to add these columns to a table and then materialize the table all at once with a single query.

Reply:
Done